File: Clojure/problem002.clj
#! /usr/bin/env clojure
; Problem 2 - Project Euler
; http://projecteuler.net/index.php?section=problems&id=2

(defn fibevensum [xmax]
  (loop [a 1 b 2 sum 0]
    (if (>= a xmax)
      sum
      (if (even? a)
        (recur b (+ a b) (+ sum a))
        (recur b (+ a b) sum)))))

(-> (fibevensum 4000000)
    (println))

File: Julia/problem002.jl
#=
Problem 2 - Project Euler
http://projecteuler.net/index.php?section=problems&id=2
=#

function fibevensum(a, b, sum, xmax)
    if a >= xmax
        sum
    elseif a % 2 == 0
        fibevensum(b, a+b, sum+a, xmax)
    else
        fibevensum(b, a+b, sum, xmax)
    end
end

println(fibevensum(1,2,0, 4000000))
